### PR TITLE
Reground sentences

### PR DESCRIPTION
**Pull request by kwalcock**

### CODE REVIEW

**Comment by kwalcock** [comment]
@zupon, this isn't ready to go yet, but the intended functionality has been placed into OntologyHandler.reground(sentenceText: String, interval: Interval).  If that looks like the wrong place or the wrong shape, you can let me know.  The idea is to go through all the steps that would normally be taken with OntologyHandler.process(eidosMention) but not have each individual grounder need to deal with anything other than an EidosMention or have each separately convert text to a Mention.

**Comment by kwalcock** [comment]
@zupon, after this is merged, you can change a couple of lines in ExtractAndExport.scala to something like

```scala
  val groundedAs = groundAs.flatMap { grounder =>
    grounder match {
      case "wm_compositional" =>
        CompositionalGrounder.branches.map(OntologyHandler.mkBranchName(grounder, Some(_))
      case other => Seq(other)
    }
  }
```

**Comment by kwalcock** [comment]
There are problems if, for example, the sentence text coming in gets parsed into two different sentences.  I'm punting on that issue here.

**Comment by zupon** [comment]
Thanks @kwalcock ! I think this should work for my purpose. @BeckySharp , does this look alright to you?

File: src/main/scala/org/clulab/wm/eidos/apps/GroundCanonicalNames.scala [hunk 1]
```diff
@@ -40,6 +42,24 @@ object GroundCanonicalNames extends App {
         (name, value, nameToIsLeaf(name))
       }
     }
+
+    def ground(text: String, canonicalName: String): Option[(String, Float, Boolean)] = {
+      val canonicalNameParts = canonicalName.split(' ')
+      val start = text.toLowerCase.indexOf(canonicalNameParts.head.toLowerCase)
+      val stop = text.toLowerCase.indexOf(canonicalNameParts.last.toLowerCase, start)
```
Comment (BeckySharp) [inline src/main/scala/org/clulab/wm/eidos/apps/GroundCanonicalNames.scala]: These seem problematic -- what if we had the last word appearing N times in a sentence, isn't `indexOf` going to return the first occurrence?

**Comment by kwalcock** [comment]
I should have written a comment.  That method is never called.  It was just used temporarily to test the call to the regrounder.  It usually worked.  For the regrounding call, something else needs to decide on the real interval.

**Comment by BeckySharp** [comment]
got it! thanks!

**Comment by kwalcock** [comment]
Some documentation has been added so that I remember what it was there for.

**Pull request merged**